src/wrappers/cache.js

```diff
@@ -2,11 +2,19 @@ import ms from 'ms';
 import debug from 'debug';
 import memoizer from 'lru-memoizer';
 
-export default function(client, { cacheMaxEntries = 5, cacheMaxAge = ms('10m') } = options) {
+export default function(client, { customCache, cacheMaxEntries = 5, cacheMaxAge = ms('10m') } = options) {
   const logger = debug('jwks');
   const getSigningKey = client.getSigningKey;
 
   logger(`Configured caching of signing keys. Max: ${cacheMaxEntries} / Age: ${cacheMaxAge}`);
+
+  if (customCache) {
```

Comment:
The cache will need to fallback to `getSigningKey` when you get a miss (and populate the cache when `getSigningKey` returns something).

Otherwise your example:

```js
client = new JwksClient({
    jwksUri: `${jwksHost}/.well-known/jwks.json`,
    cache: true,
    customCache: new CustomCache()
  });
```

Will never resolve anything

Reply:
Ideally a custom cache should replace memoizer (the default in-memory cache) rather than sit in front of it.

Reply:
Perhaps using something like `cache-manager`, eg

```js
import cacheManager from 'cache-manager';

export default function(client, { cacheMaxEntries = 5, cacheMaxAge = ms('10m'), store = 'memory' } = options) {
  const memoryCache = cacheManager.caching({ store, max: cacheMaxEntries, ttl: cacheMaxAge });
  return (kid, cb) => memoryCache.wrap(kid, (cacheCallback) => {
    client.getSigningKey(kid, cacheCallback);
  }, {}, cb);
}
```

You'd ideally need to provide an lru cache as your custom cache, then the `cacheMaxEntries` and `cacheMaxAge` apply to the custom cache as well. eg https://github.com/BryanDonovan/node-cache-manager#custom-stores